torch/csrc/jit/codegen/cuda/parser.cpp

```diff
@@ -613,116 +616,359 @@ class IrParser {
           });
     }
 
+    {
```

Comment:
@rdspring1 Would you mind to take a quick look here since I copied most of this from your old PR.

Reply:
note, this should have been cleaned up further. We briefly talked about putting some higher level pattern in nn_arith.cpp. We can revisit this then.